src/main/java/org/opensearch/security/user/InternalUserProvider.java

```diff
@@ -0,0 +1,20 @@
+package org.opensearch.security.user;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.io.IOException;
+import org.opensearch.security.securityconf.impl.SecurityDynamicConfiguration;
+
+/**
+ * This interface will be defined in core following these changes. For now this is used for testing in the Security Plugin.
+ *
+ */
+public interface InternalUserProvider {
+
+    public void putInternalUser(String userInfo) throws java.io.IOException;
+
+    public SecurityDynamicConfiguration<?> getInternalUser(String userInfo);
+
+    public void removeInternalUser(String userInfo);
+
+    public String getInternalUserAuthToken(String userInfo) throws IOException;
```

Comment:
I don't think this return type should be a string in OpenSearch, it should be some kind of interface around the auth token.  @cwperks Did you have ideas about how we would pass refresh token in the java ecosystem?